metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -899,6 +899,77 @@ def _extract_lineages(self):
         return
 
 
+@dataclass
+class ConfluentS3SinkConnector:
+    connector_manifest: ConnectorManifest
+
+    def __init__(
+        self, connector_manifest: ConnectorManifest, report: KafkaConnectSourceReport
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.report = report
+        self._extract_lineages()
+
+    @dataclass
+    class S3SinkParser:
+        target_platform: str
+        bucket: str
+        topics_dir: str
+        topics: List[str]
+
+    def _get_parser(self, connector_manifest: ConnectorManifest) -> S3SinkParser:
+        # https://docs.confluent.io/kafka-connectors/s3-sink/current/configuration_options.html#s3
+        bucket = connector_manifest.config.get("s3.bucket.name")
+        if not bucket:
+            raise ValueError(f"Could not find 's3.bucket.name' in connector configuration {self.connector_manifest.name}")
+
+        # https://docs.confluent.io/platform/current/installation/configuration/connect/sink-connect-configs.html#topics
+        topics = connector_manifest.config.get("topics")
```

Comment:
The sink config may not contain `topics` config if `topics.regex` config is provided.  You may use `connector_manifest.topic_names` here. Its should already be populated as the source calls `/topics` rest endpoint for connector in advance.

Reply:
Thanks, I’ll fix that.